design/ceph/object/ceph-cosi-driver.md

```diff
@@ -0,0 +1,142 @@
+
+```bash
+kubectl create -f deploy/examples/cosi/bucketclass.yaml
+kubectl create -f deploy/examples/cosi/bucketclaim.yaml
+kubectl create -f deploy/examples/cosi/bucketaccessclass.yaml
+kubectl create -f deploy/examples/cosi/bucketaccess.yaml
+```
+
+#### Consuming COSI Bucket
+
+The user needs to mount the secret as volume created by `BucketAccess` to the application pod. The user can use the secret to access the bucket by parsing the mounted file.
+
+```yaml
+spec:
+  containers:
+      volumeMounts:
+        - name: cosi-secrets
+          mountPath: /data/cosi
+  volumes:
+  - name: cosi-secrets
+    secret:
+      secretName: ba-secret
+```
+
+```bash
+# cat /data/cosi/bucket_info.json
+
+    |-----------------------------------------------|
+    | {                                             |
+    |   apiVersion: "v1alpha1",                     |
+    |   kind: "BucketInfo",                         |
+    |   metadata: {                                 |
+    |       name: "ba-$uuid"                        |
+    |   },                                          |
+    |   spec: {                                     |
+    |       bucketName: "ba-$uuid",                 |
+    |       authenticationType: "KEY",              |
+    |       endpoint: "https://s3.amazonaws.com",   |
+    |       accessKeyID: "AKIAIOSFODNN7EXAMPLE",    |
+    |       accessSecretKey: "wJalrXUtnFEMI/K...",  |
+    |       region: "us-west-1",                    |
+    |       protocols: [                            |
+    |         "s3"                                  |
+    |       ]                                       |
+    |   }                                           |
+    | }                                             |
+    |-----------------------------------------------|
+
+```
+
+#### Coexistence of COSI and libbucket provisioner
+
+Currently the ceph object store provisioned via Object Bucket Claim (OBC). They both can co exist and can even use same backend bucket from ceph storage. No deployment/configuration changes are required to support both. The libbucket probvisioner is deprecated and eventually will be replaced by COSI when it becomes more and more stable. The CRDs used by both are different hence there is no conflicts between them.
+
+#### Transition from libbucket provisioner to COSI
+
+This applied to OBC with reclaim policy is `Retain`. First the user need to create a COSI bucket pointing to the backend bucket. Then user can create `bucket access class` and `bucket access` using the `COSI bucket`. Now the update applications credentials with bucket access credentials, finally the user need to delete the OBC.
```

Comment:
When the OBC is deleted, is there a secret that will also be deleted? So their application would be broken unless they also update their application to read from the COSI file `/data/cosi/bucket_info.json`, correct?